be/src/vec/exec/format/parquet/vparquet_group_reader.cpp

```diff
@@ -599,6 +668,236 @@ Status RowGroupReader::_filter_block_internal(Block* block,
+        }
+        {
+            TExprNode texpr_node;
+            texpr_node.__set_node_type(TExprNodeType::INT_LITERAL);
+            texpr_node.__set_type(create_type_desc(TYPE_INT));
+            TIntLiteral int_literal;
+            int_literal.__set_value(dict_codes[0]);
+            texpr_node.__set_int_literal(int_literal);
+            VExpr* literal_expr = _obj_pool->add(new VLiteral(texpr_node));
+            root->add_child(literal_expr);
+        }
+    } else {
+        {
+            TTypeDesc type_desc = create_type_desc(PrimitiveType::TYPE_BOOLEAN);
+            TExprNode node;
+            node.__set_type(type_desc);
+            node.__set_node_type(TExprNodeType::IN_PRED);
+            node.in_predicate.__set_is_not_in(false);
+            node.__set_opcode(TExprOpcode::FILTER_IN);
+            node.__isset.vector_opcode = true;
+            node.__set_vector_opcode(TExprOpcode::FILTER_IN);
+
+            root = _obj_pool->add(new vectorized::VDirectInPredicate(node));
+            std::shared_ptr<HybridSetBase> hybrid_set(create_set(PrimitiveType::TYPE_INT));
+            for (int j = 0; j < dict_codes.size(); ++j) {
+                hybrid_set->insert(&dict_codes[j]);
+            }
+            static_cast<vectorized::VDirectInPredicate*>(root)->set_filter(hybrid_set);
+        }
+        {
+            SlotDescriptor* slot = nullptr;
+            const std::vector<SlotDescriptor*>& slots = _tuple_descriptor->slots();
+            for (auto each : slots) {
+                if (each->id() == slot_id) {
+                    slot = each;
+                    break;
+                }
+            }
+            VExpr* slot_ref_expr = _obj_pool->add(new VSlotRef(slot));
+            root->add_child(slot_ref_expr);
+        }
+    }
+    VExprContext* rewritten_conjunct_ctx = _obj_pool->add(new VExprContext(root));
+    RETURN_IF_ERROR(rewritten_conjunct_ctx->prepare(_state, *_row_descriptor));
+    RETURN_IF_ERROR(rewritten_conjunct_ctx->open(_state));
+    _dict_filter_conjuncts.push_back(rewritten_conjunct_ctx);
+    _filter_conjuncts.push_back(rewritten_conjunct_ctx);
+    return Status::OK();
+}
+
+void RowGroupReader::_convert_dict_cols_to_string_cols(Block* block) {
+    for (auto& dict_filter_col_name : _dict_filter_col_names) {
+        ColumnPtr& column = block->get_by_name(dict_filter_col_name).column;
+        if (auto* nullable_column = check_and_get_column<ColumnNullable>(*column)) {
+            const ColumnPtr& nested_column = nullable_column->get_nested_column_ptr();
+            const ColumnDictI32* dict_column =
+                    assert_cast<const ColumnDictI32*>(nested_column.get());
+            DCHECK(dict_column);
+
+            MutableColumnPtr string_column =
+                    _column_readers[dict_filter_col_name]->convert_dict_column_to_string_column(
+                            dict_column);
+
+            size_t pos = block->get_position_by_name(dict_filter_col_name);
```

Comment:
869 line search hash map by name， maybe better call `size_t pos = block->get_position_by_name(dict_filter_col_name);` in  869, only search hash map once time